dspace-api/src/main/java/org/dspace/identifier/doi/DOIOrganiser.java

```diff
@@ -245,8 +252,16 @@ public static void runCLI(Context context, DOIOrganiser organiser, String[] args
                                            + "that could be registered.");
                 }
                 for (DOI doi : dois) {
-                    organiser.register(doi);
-                    context.uncacheEntity(doi);
+                    doi = context.reloadEntity(doi);
+                    try {
+                        organiser.register(doi);
+                        context.commit();
+                    } catch (SQLException e) {
+                        System.err.format("DOI %s for object %s registration failed, skipping:  %s%n",
+                                doi.getDSpaceObject().getID().toString(),
+                                doi.getDoi(), e.getMessage());
+                        context.rollback();
+                    }
```

Comment:
The `register` method can throw  `IllegalStateException | RuntimeException` that are not catched inside this block. Those exception would block the whole process.

Reply:
Thanks for catching that.  Ugh, unchecked exceptions.  Undocumented in `register`'s signature, too.  I'll fix that.